syft/frameworks/torch/federated/dataset.py

```diff
@@ -60,14 +61,9 @@ def transform(self, transform):
         """
 
         # Transforms cannot be applied to Pointer  , Fixed Precision or Float Precision tensors.
-        if type(self.data) == torch.Tensor:
```

Comment:
I would prefer to keep it this way with an additional elif statement for checking numpy type and applying the transform. Since using datatypes we haven't explicitly tested could lead to untraceble errors. 

So
`if type(self.data) == torch.Tensor:
        self.data = transform(self.data)
  else type(self.data)==<the numpy format>:
        self.data=transform(self.data)
   else:
         raise TypeError("Transforms can be applied only on torch tensors")
 `

Reply:
Then how would be differentiate between transforms, one way is to apply transforms independent of datatype if is either torch.tensor or numpy array. e.g
```
if type(self.data) == torch.Tensor or type(self.data) == numpy.ndarray:
    self.data = transform(self.data)
else: raise TypeError("Transforms can be applied only on torch tensors or numpy array")
```